src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        if (this.factory == null) {
+            throw new DataProcessingException("Factory not initialized");
+        }
+        Session session = this.factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
+        Transaction transaction = session.beginTransaction();
+        try {
+            session.persist(country);
+            transaction.commit();
+        } catch (Exception e) {
+            transaction.rollback();
+            throw new DataProcessingException("Error in adding actor", e);
```

Comment:
The exception message should be specific to the `CountryDaoImpl` class. The current message mentions adding an 'actor' instead of a 'country'. Please correct the entity type in the exception message.